public/app/features/dashboard/services/TimeSrv.ts

```diff
@@ -276,20 +276,19 @@ export class TimeSrv {
     this.setAutoRefresh(this.previousAutoRefresh);
   }
 
-  setTime(time: RawTimeRange, fromRouteUpdate?: boolean) {
+  setTime(time: RawTimeRange, updateUrl = true) {
     extend(this.time, time);
 
     // disable refresh if zoom in or zoom out
     if (isDateTime(time.to)) {
-      this.oldRefresh = this.dashboard?.refresh || this.oldRefresh;
+      this.oldRefresh = this.timeModel?.refresh || this.oldRefresh;
       this.setAutoRefresh(false);
-    } else if (this.oldRefresh && this.oldRefresh !== this.dashboard?.refresh) {
+    } else if (this.oldRefresh && this.oldRefresh !== this.timeModel?.refresh) {
       this.setAutoRefresh(this.oldRefresh);
       this.oldRefresh = null;
     }
 
-    // update url
-    if (fromRouteUpdate !== true) {
+    if (updateUrl === true) {
```

Comment:
@torkelo I changed the logic in the function to be inverted as well. Is there a bug with this I can look at?

Reply:
@gelicia yea, https://github.com/grafana/grafana/pull/48657 

But no worries will be fixed in the next patch release. Wish we had better unit tests for this part (TimeSrv is very messy, should be refactored to redux actions)

Reply:
Thank you for fixing my bug 😅